### PR TITLE
digital media returns key

### PR DESCRIPTION
**Pull request by southeo**
- Returns digitalMediaKey, an object that contains specimenID and url 
- specify local id type so we can choose what kind of indexed id we get from mongo

### CODE REVIEW

File: src/main/java/eu/dissco/core/handlemanager/configuration/KafkaPublisherConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+
+@Configuration
+@AllArgsConstructor
+@Profile(Profiles.DOI)
+public class KafkaPublisherConfiguration {
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/configuration/KafkaPublisherConfiguration.java:20]: Yeah that's what it's needed for. Testing the handle api alone, I didn't notice this was missing. When I tested the integration with the specimen processor and media processor, with other kafka messages being sent, kafka threw an error and I realized it was missing some configuration. I can make it a separate PR but it's just copied from the other kafka configurations we have.

File: src/main/java/eu/dissco/core/handlemanager/service/PidService.java [hunk 2]
```diff
@@ -405,4 +413,11 @@ protected void updateDocuments(List<FdoRecord> fdoRecords)
     mongoRepository.updateHandleRecords(fdoDocuments);
     log.info("Successfully updated {} specimens fdo records to database", fdoDocuments.size());
   }
+
+  private record DigitalMediaKey(
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/service/PidService.java:417]: it's used to format one response, I just refactored the function it was used in

[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/handlemanager/service/PidService.java]: Oeh so I might have put this in, note to self then...

**Review by samleeflang — APPROVED** [review]
:100:

**Pull request merged**